backend/app/tests.py

```diff
@@ -119,10 +119,119 @@ class MainTests(TestCase):
+        doc_2.save()
+        tokenized_text_2 = ['she', 'really', 'likes', 'to', 'eat', 'chocolate']
+        self.assertEqual(doc_1.get_tokenized_text(), tokenized_text_1)
+        self.assertEqual(doc_2.get_tokenized_text(), tokenized_text_2)
+
+    def test_get_word_count(self):
+        doc_1 = Document(text='The quick brown fox jumped over the lazy dog.')
+        doc_1.save()
+        word_count_1 = 9
+        doc_2 = Document(text='She really likes to eat chocolate!')
+        doc_2.save()
+        word_count_2 = 6
+        self.assertEqual(doc_1.get_word_count(), word_count_1)
+        self.assertEqual(doc_2.get_word_count(), word_count_2)
+
+    def test_get_wordcount_counter(self):
+        doc_1 = Document(text='Do you like ice cream as much as I do?')
+        doc_1.save()
+        counter_1 = Counter({'do': 2, 'you': 1, 'like': 1, 'ice': 1, 'cream': 1, 'as': 2, 'much': 1, 'i': 1})
+        doc_2 = Document(text='She really likes to eat chocolate!')
+        doc_2.save()
+        counter_2 = Counter({'she': 1, 'really': 1, 'likes': 1, 'to': 1, 'eat': 1, 'chocolate': 1})
+        self.assertEqual(doc_1.get_wordcount_counter(), counter_1)
+        self.assertEqual(doc_2.get_wordcount_counter(), counter_2)
+
+    def test_clean_quotes(self):
+        doc_1 = Document(text='This is a “smart” phrase')
+        doc_1.save()
+        cleaned_1 = 'This is a "smart" phrase'
+        self.assertEqual(doc_1._clean_quotes(), cleaned_1)
+
+    def test_find_quoted_text(self):
+        doc_1 = Document(text='"This is a quote." There is more. "This is my quote."')
+        doc_1.save()
+        quoted_text_1 = ['"This is a quote."', '"This is my quote."']
+        self.assertEqual(doc_1.find_quoted_text(), quoted_text_1)
+
+    def test_get_count_of_word(self):
+        doc_1 = Document(text='Do you like ice cream as much as I do?')
+        doc_1.save()
+        self.assertEqual(doc_1.get_count_of_word('do'), 2)
+        self.assertEqual(doc_1.get_count_of_word('ThisWordIsNotThere'), 0)
+
+    def test_get_count_of_words(self):
+        doc_1 = Document(text='Do you like ice cream as much as I do?')
+        doc_1.save()
+        self.assertEqual(doc_1.get_count_of_words(['do', 'as', 'you']), Counter({'do': 2, 'as': 2, 'you': 1}))
+        self.assertEqual(doc_1.get_count_of_words(['ThisWordIsNotThere', 'well']), Counter({'ThisWordIsNotThere': 0, 'well': 0}))
+
+    def test_words_associated(self):
+        doc_1 = Document(text="""She took a lighter out of her purse and handed it over to him.
+                                 He lit his cigarette and took a deep drag from it, and then
+                                 began his speech which ended in a proposal. Her tears drowned the ring.
+                                 TBH i know nothing about this story.""")
+        doc_1.save()
+        self.assertEqual(doc_1.words_associated('his'), Counter({'cigarette': 1, 'speech': 1}))
```

Comment:
Also consider making a test case with "her" because it also appears a few times in this document.